### DISCUSSION TITLE
Model Serializers and Multiple Entries

### DISCUSSION THREAD

**Discussion by patagoniapy** [encode/django-rest-framework#8542] [General]
I recently stumbled upon what I thought would be a fairly common use case, and therefore supported by DRF out of the box, but this wasn't the case (unless I'm doing something wrong, which is very likely) 

Here's an overview 

I had the model `Transaction Detail`
```python
class TransactionDetail(models.Model):
    transaction_id = models.ForeignKey(Transaction, on_delete=models.CASCADE)
    amount = models.FloatField()
    account_name = models.ForeignKey(
        Account, on_delete=models.CASCADE)
```
Which I was serializing using `ModelSerializer`
```python
class TransactionDetailSerializer(serializers.ModelSerializer):
    class Meta:
        model = TransactionDetail
        fields = ['amount', 'account_name']
```
And accessing with this view
```python
class TransactionDetailViewSet(mixins.CreateModelMixin,
                               mixins.ListModelMixin,
                               mixins.RetrieveModelMixin,
                               viewsets.GenericViewSet):


    queryset = TransactionDetail.objects.all()
    serializer_class = TransactionDetailSerializer

    def perform_create(self, serializer):
       """
         Custom logic for TransactionID here that all works
       """
```
All was well if I was submitting one `Transaction Detail`, but as soon as I tried to pass something like below, I got the error `Expected a dictionary, but got a list`
```json
[
    {
        "account_name": 1,
        "amount": 15,
    },
    {
        "account_name": 2,
        "amount": 15,
    }
]
```
I was able to solve this by  following [this](https://stackoverflow.com/questions/48870489/make-django-rest-api-accept-a-list/48885641#48885641) thread
```python
class CreateListModelMixin(object):
    def get_serializer(self, *args, **kwargs):
        """ if an array is passed, set serializer to many """
        if isinstance(kwargs.get('data', {}), list):
            kwargs['many'] = True
        return super(CreateListModelMixin, self).get_serializer(*args, **kwargs)
```
My first question is if I'm missing anything that would allow DRF to behave this way out of the box?

If not, then my second question is why isn't this the default behavior in DRF? It's hard for me to see the downside in running this logic, and the amount of questions about this on StackOverflow lead me to believe it's a fairly common application.

**Comment by Azanul** [comment]
The mixins actually provide methods independently from one another.
This functionality would make sense only if ```CreateModelMixin``` and ```ListModelMixin``` both are extended. Such functionality out of the box would mess with the independent part.
Also, I think this is not supplied out of the box is because many people want to accept single objects at once only. People who want to accept a list of entries for creating can extend in all sorts of ways (as you have).
